cli/npm/resolution/graph.rs

```diff
@@ -900,134 +914,159 @@ impl<'a> GraphDependencyResolver<'a> {
-            // only necessary for graph segments that could potentially have peer
-            // dependencies within them.
-            continue;
-          }
+    // go down through the dependencies by tree depth
+    while let Some(parent_path) = self.pending_unresolved_nodes.pop_front() {
+      let (parent_nv, child_deps) = {
+        let node_id = parent_path.node_id();
+        if self.graph.nodes.get(&node_id).unwrap().no_peers {
+          // We can skip as there's no reason to analyze this graph segment further.
+          continue;
+        }
 
-          let pkg_nv = self
-            .graph
-            .resolved_node_ids
-            .get(node_id)
-            .unwrap()
-            .nv
-            .clone();
-          let deps = if let Some(deps) = self.dep_entry_cache.get(&pkg_nv) {
-            deps.clone()
-          } else {
-            // the api should have this in the cache at this point, so no need to parallelize
-            match self.api.package_version_info(&pkg_nv).await? {
-              Some(version_info) => {
-                self.dep_entry_cache.store(pkg_nv.clone(), &version_info)?
-              }
-              None => {
-                bail!("Could not find version information for {}", pkg_nv)
-              }
+        let pkg_nv = self
+          .graph
+          .resolved_node_ids
+          .get(node_id)
+          .unwrap()
+          .nv
+          .clone();
+        let deps = if let Some(deps) = self.dep_entry_cache.get(&pkg_nv) {
+          deps.clone()
+        } else {
+          // the api should have this in the cache at this point, so no need to parallelize
+          match self.api.package_version_info(&pkg_nv).await? {
+            Some(version_info) => {
+              self.dep_entry_cache.store(pkg_nv.clone(), &version_info)?
             }
-          };
-
-          (pkg_nv, deps)
+            None => {
+              bail!("Could not find version information for {}", pkg_nv)
+            }
+          }
         };
 
-        // cache all the dependencies' registry infos in parallel if should
-        self
-          .api
-          .cache_in_parallel({
-            deps.iter().map(|dep| dep.name.clone()).collect()
-          })
-          .await?;
-
-        // resolve the dependencies
-        let mut found_peer = false;
-
-        for dep in deps.iter() {
-          let package_info = self.api.package_info(&dep.name).await?;
-
-          match dep.kind {
-            NpmDependencyEntryKind::Dep => {
-              // todo(dsherret): look into skipping dependency analysis if
-              // it was done previously again
-              let child_id =
-                self.analyze_dependency(dep, &package_info, &graph_path)?;
+        (pkg_nv, deps)
+      };
 
-              if !found_peer {
-                found_peer = !self.graph.borrow_node_mut(child_id).no_peers;
+      // cache all the dependencies' registry infos in parallel if should
```

Comment:
This comment makes no sense :(

Reply:
I'm not sure how to reword this because it's saying what the next line does